torch/csrc/autograd/profiler.h

```diff
@@ -131,15 +156,51 @@ struct TORCH_API Event final {
       uint16_t thread_id,
       bool record_cuda,
       at::RecordFunctionHandle handle = 0,
-      std::vector<std::vector<int64_t>>&& shapes = {})
+      std::vector<std::vector<int64_t>>&& shapes = {}, int node_id = -1, double relative_elapsed_cpu_us = -1)
       : name_(std::move(name)),
         kind_(kind),
         thread_id_(thread_id),
         handle_(handle),
-        shapes_(shapes) {
+        shapes_(shapes), node_id_(node_id), relative_elapsed_cpu_us_(relative_elapsed_cpu_us) {
     record(record_cuda);
   }
 
+  // Returns IValues corresponding to event structure, to be used for
+  // serialization.
+  at::IValue toIValues() const {
+    c10::impl::GenericList eventIValueList(at::AnyType::get());
```

Comment:
Does it worth it to use `reserve()` and then `emplace_back`?

Reply:
Good call, updating